### PR TITLE
Add recipe for `cargo-llvm-cov`

### PR DESCRIPTION
**Pull request by borchero**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by borchero** [comment]
Agree to be listed as maintainer ✅

File: recipes/cargo-llvm-cov/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: 0a1b63922cfb6f2526a61bc5e13868c3f3345aa3af7337df07a69eab3d4b243e
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
+    - {{ compiler('c') }}        # [unix]
+    - {{ compiler('m2w64_c') }}  # [win]
```
Comment (h-vetinari) [inline recipes/cargo-llvm-cov/meta.yaml:18]: Any particular reason why you want the m2 compilers on windows? Why not simply:
```suggestion
    - {{ compiler('c') }}
```

Comment (borchero) [reply recipes/cargo-llvm-cov/meta.yaml:18]: Not really, no. I copied this revipe from taplo and, once it worked, didn't bother to find the minimum set of build dependencies 😬 I can try removing it 👍🏼

File: recipes/cargo-llvm-cov/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: 0a1b63922cfb6f2526a61bc5e13868c3f3345aa3af7337df07a69eab3d4b243e
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
+    - {{ compiler('c') }}  # [unix]
+    - {{ stdlib('c') }}    # [unix]
```
Comment (h-vetinari) [inline recipes/cargo-llvm-cov/meta.yaml:18]: Why do we need a C-compiler on unix but not windows? Does the package do anything differently between unix/win?

It's worth noting that the rust compiler brings a C-compiler in any case; similarly, stdlib should be present everywhere.

File: recipes/cargo-llvm-cov/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - cargo-bundle-licenses
+  host:
+    - clangdev  # [not osx]
+    - llvmdev   # [not osx]
```
Comment (xhochy) [inline recipes/cargo-llvm-cov/meta.yaml:22]: I would suspect that we also need them on `osx` but simply link against the compilers that are installed in `build`. Also, do you know which parts of `llvm` are used?

Comment (borchero) [reply recipes/cargo-llvm-cov/meta.yaml]: Seems like these host dependencies are, in fact, not required. I originally copied this recipe from [taplo](https://github.com/conda-forge/taplo-feedstock) to get started and I was happy once it built, hence, didn't try to find the minimal set of build dependencies :eyes:

File: recipes/cargo-llvm-cov/meta.yaml (hunk 2 above)
Comment (borchero) [reply recipes/cargo-llvm-cov/meta.yaml]: Without the compiler, I'm getting the following error:

```
Traceback (most recent call last):
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 261, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 151, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 207, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.10/site-packages/conda_build/api.py", line 250, in build
    return build_tree(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 3762, in build_tree
    packages_from_this = build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 2839, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 1974, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 1782, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/post.py", line 1729, in post_build
    check_overlinking(m, files, host_prefix)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/post.py", line 1554, in check_overlinking
    return check_overlinking_impl(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/post.py", line 1531, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (cargo-llvm-cov,bin/cargo-llvm-cov): $RPATH/libgcc_s.so.1 not found in packages, sysroot(s) nor the missing_dso_whitelist.\n.. is this binary repackaging?']
```

Comment (h-vetinari) [reply recipes/cargo-llvm-cov/meta.yaml]: In my opinion this is an [issue](https://github.com/conda-forge/rust-activation-feedstock/issues/51) with the rust activation that should be fixed there, once, and not worked around in many rust-recipes by adding `{{ compiler("c") }}  # [linux]`.

**Pull request merged**